### PR TITLE
Decode FooterData of model texture

### PR DESCRIPTION
**Pull request by kenjiuno**
Decode FooterData of model texture.
`OpenKh.Research.MapTexFooter` command line tool is a sample usage of `TextureFooterData`.

```cs
                var modelTexture = ModelTexture.Read(entry.Stream);

                var footerData = modelTexture.TextureFooterData;

                Console.WriteLine(
                    new SerializerBuilder()
                        // omit unimportant and massive array elements.
                        .WithAttributeOverride<TextureAnimation>(it => it.SpriteImage, new YamlIgnoreAttribute())
                        .WithAttributeOverride<TextureAnimation>(it => it.SlotTable, new YamlIgnoreAttribute())
                        .Build()
                        .Serialize(footerData)
                );
```

Doc is ready:
- https://github.com/kenjiuno/OpenKh/blob/tex-footer/docs/tool/CLI.TexFooter/index.md

### CODE REVIEW

**Comment by Xeeynamo** [comment]
Excited for this PR :)

**Comment by Xeeynamo** [comment]
With very little effort, this is what I was able to achieve thanks to your parser:
texscroll

I am still reviewing, but so far it's very exciting!

File: OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        [Required]
+        [FileExists]
+        [Argument(0, Description = "Map file")]
+        public string MapFile { get; set; }
+
+        [Argument(1, Description = "Output dir")]
+        public string OutputDir { get; set; }
+
+        protected int OnExecute(CommandLineApplication app) => OnExecute();
```
Comment (Xeeynamo) [inline OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs:31]: Unfortunately having a second `OnExecute` makes the application crashing

File: OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        {
+            var list = new List<string>();
+
+            var barEntries = File.OpenRead(MapFile).Using(Bar.Read);
+            var entry = barEntries.FirstOrDefault(entry => entry.Type == Bar.EntryType.ModelTexture);
+            if (entry == null)
+            {
+                return 1;
+            }
+            if (entry.Stream.Length < 0x60)
```
Comment (Xeeynamo) [inline OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs:43]: Maybe a good idea would be to use `ModelTexture.IsValid(stream)` instead?

**Review by Xeeynamo — CHANGES_REQUESTED** [review]
I have few request changes to suggest before approving it:

* The `TextureFooterData` is not tested. There is a `ShouldWriteBackTheExactSameFile`, but the decorator to run the unit test is commented. Would you be able to use a custom texture (to not upload copyright stuff) that will be used for that test?
* The tool Command.TexFooter crashes with the command `export`. I pointed where the problem is.
* Exporting the footer from maps works, but it only works for the first texture it finds. Some maps have `MAP`, `SK0` and `SK1`, but the tool deserialize only the first one that is found.

A small note for the future: do you think would be a good idea to always deserialize the footer in `ModelTexture` object? Because in theory we always want to read the footer after reading the image. It's not a require change but it might be something to consider in the future.

File: OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs (hunk 1 above)
Comment (kenjiuno) [reply OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs]: Thanks, fixed!

File: OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs (hunk 2 above)
Comment (kenjiuno) [reply OpenKh.Command.TexFooter/Subcommands/ExportCommand.cs]: Thanks, `IsValid` added!

**Comment by kenjiuno** [comment]
Sorry I have missed comments on this pull request. I have updated the corresponding codes.

**Review by Xeeynamo — APPROVED** [review]

**Pull request merged**